todo_app/tests/test_client.py

```diff
@@ -1,53 +1,32 @@
 import json
 from dotenv import load_dotenv, find_dotenv
 from flask.testing import FlaskClient
-import pytest
 import mongomock
+import requests
+import pytest
+from pytest import MonkeyPatch
 from todo_app import app
-
+from flask import Flask, redirect, url_for
+from flask_dance.consumer.storage import MemoryStorage
+from todo_app.oauth import blueprint
+from flask_dance.contrib.github import github
 
 @pytest.fixture
-def client():
+def client(monkeypatch: pytest.MonkeyPatch):
+    # Use our test integration config instead of the 'real' version
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
+    storage = MemoryStorage({"access_token": "client_secret"})
+    monkeypatch.setattr(blueprint, 'storage', storage)      
     with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+    # Create the new app.
         test_app = app.create_app()
-        with test_app.test_client() as client:
-            yield client
-            
-
-class StubResponse():
-    def __init__(self, fake_response_data):
-        self.fake_response_data = fake_response_data
-        self.content=json.dumps(fake_response_data)
-
-    def json(self):
-        return self.fake_response_data
-
-# Stub replacement for requests.get(url)
-def stub(url, params={}):
-    fake_response_data = None
-    if url == f'https://localhost:5000':
-        fake_response_data = [{
-            '_id': '123abc',
-            'name': 'To Do',
-            'desc': 'A basic To Do item',
-            'Status': 'To Do'
-        },{'_id': '123abcd',
-            'name': 'Doing',
-            'desc': 'A basic To Do item',
-            'Status': 'In Progress'}
-        ,{'_id': '123abcde',
-            'name': 'Done',
-            'desc': 'A basic To Do item',
-            'Status': 'Completed'}]
-        return StubResponse(fake_response_data)
-    raise Exception(f'Integration test did not expect URL "{url}"')
-
+    # Use the app to create a test_client that can be used in our tests.
+    with test_app.test_client() as client:
+        yield client
```

Comment:
We can mock out the session storage here for our testing to get this back to passing:
```suggestion
    with test_app.test_client() as client:
        with client.session_transaction() as session:
            session["username"] = "smellybeagle"
        yield client
```

(Taken from this [SO discussion](https://stackoverflow.com/questions/36509662/python-mock-testing-to-mock-session))